applications/nrf_desktop/doc/dfu_mcumgr.rst

```diff
@@ -0,0 +1,88 @@
+  For details, see :ref:`zephyr:mcu_mgr` in the Zephyr documentation.
+* MCUmgr groups:
+
+  * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG` - This option enables the MCUmgr image management handlers, which are required for the DFU process.
+    For details, see :ref:`zephyr:device_mgmt` in the Zephyr documentation.
+  * :kconfig:option:`CONFIG_MCUMGR_GRP_OS` - This option enables the MCUmgr OS management handlers, which are required for the DFU process.
+    For details, see :ref:`zephyr:device_mgmt` in the Zephyr documentation.
+
+* :kconfig:option:`CONFIG_MCUMGR_MGMT_NOTIFICATION_HOOKS` - This option enables the MCUmgr notification hook support, which allows the module to listen for an MCUmgr event.
+  For details, see :ref:`zephyr:mcumgr_callbacks` in the Zephyr documentation.
+
+  MCUmgr notifications hooks:
+
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_UPLOAD_CHECK_HOOK` - This option enables the MCUmgr upload check hook, which sends image upload requests to the registered callbacks.
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_STATUS_HOOKS` - This option enables the MCUmgr image status hooks, which report the DFU status to the registered callbacks.
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_OS_RESET_HOOK` - This option enables the MCUmgr OS reset hook, which sends reset requests to the registered callbacks.
+
+* MCUmgr dependencies:
+
+  * :kconfig:option:`CONFIG_NET_BUF`
+  * :kconfig:option:`CONFIG_ZCBOR`
+  * :kconfig:option:`CONFIG_CRC`
+  * :kconfig:option:`CONFIG_IMG_MANAGER`
+  * :kconfig:option:`CONFIG_STREAM_FLASH`
+  * :kconfig:option:`CONFIG_FLASH_MAP`
+  * :kconfig:option:`CONFIG_FLASH`
+
+To use the module, you must also enable the :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` Kconfig option.
+The DFU over Simple Management Protocol in Zephyr is supported only with the MCUboot bootloader.
+
+You must also enable the preferred transport for the MCUmgr's SMP protocol (for example, the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT` Kconfig option).
+With the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT` Kconfig option enabled, the module enables the :kconfig:option:`CONFIG_CAF_BLE_SMP_TRANSFER_EVENTS` event.
+The event can be used to lower Bluetooth connection latency during the DFU process.
+
+The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (for example, the :ref:`nrf_desktop_dfu`).
+Set the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` Kconfig option to enable this feature.
+Make sure that the DFU lock utility is enabled if your nRF Desktop application configuration uses multiple DFU transports.
+
+You cannot use this module with the :ref:`caf_ble_smp`.
+In other words, you cannot simultaneously enable the :ref:`CONFIG_DESKTOP_DFU_MCUMGR_ENABLE <config_desktop_app_options>` option and the :kconfig:option:`CONFIG_CAF_BLE_SMP` Kconfig option.
+
+Implementation details
+**********************
+
```

Comment:
The module uses MCUmgr's image notification callbacks for the following purposes:

* To periodically submit an :c:struct:`ble_smp_transfer_event` while image upload is in progress.
   The module registers itself as the final subscriber of the event to track the number of submitted events.
   If a :c:struct:`ble_smp_transfer_event` was already submitted, but was not yet processed, the module desists from submitting subsequent event.
* To reject image upload or system reboot request if :ref:`nrf_desktop_dfu_lock` is already taken by another DFU transport.

Reply:
The `img_mgmt_reset_upload` function that is part of the MCUmgr internal API is used to clear state of the image upload if another DFU transport claims the DFU lock.

Reply:
thanks